src/components/dictionaryConcepts/components/DescriptionTable.jsx

```diff
@@ -1,37 +1,29 @@
-import React from 'react';
+import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
 import DescriptionRow from './DescriptionRow';
 
-const DescriptionTable = props => (
-  <table className="table table-striped table-bordered concept-form-table">
-    <thead className="header text-white">
-      <tr>
-        <th scope="col">Description</th>
-        <th scope="col">Language</th>
-        <th scope="col">Actions</th>
-      </tr>
-    </thead>
-    <tbody>
-      {props.existingConcept && props.existingConcept.descriptions
-        ? props.existingConcept.descriptions.map(newRow => (
-          <DescriptionRow
-            newRow={newRow}
-            rowId={newRow.uuid}
-            key={newRow.uuid}
-            {...props}
-          />
-        ))
-        : props.description.map(newRow => (
-          <DescriptionRow
-            newRowUuid={newRow}
-            rowId={newRow}
-            key={newRow}
-            {...props}
-          />))
-      }
-    </tbody>
-  </table>
-);
+class DescriptionTable extends PureComponent {
```

Comment:
Any reason this had to be changed from a functional to a pure component?

Reply:
It avoids unnecessary re-rendering of the component - a feature that pure components have which functional components don't. There was a bug where occasionally, edit page would render with more than one empty description row on refreshing the page, and I noticed they had exactly the same props.
Pure components only trigger a re-render if the props passed to it have actually changed